packages/shared/src/components/squads/SquadMemberActions.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { ReactElement } from 'react';
+import { SourceMember, SourceMemberRole } from '../../graphql/sources';
+import { Button, ButtonSize } from '../buttons/Button';
+import BlockIcon from '../icons/Block';
+import MenuIcon from '../icons/Menu';
+import { SimpleTooltip } from '../tooltips/SimpleTooltip';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { SquadMemberBadge } from './SquadMemberRoleBadge';
+import { usePrompt } from '../../hooks/usePrompt';
+import { UserShortInfo } from '../profile/UserShortInfo';
+import { ModalSize } from '../modals/common/types';
+
+interface SquadMemberActionsProps {
+  member: SourceMember;
+  onUnblock: React.MouseEventHandler;
+  onOptionsClick: React.MouseEventHandler;
+}
+
+function SquadMemberActions({
+  member,
+  onUnblock,
+  onOptionsClick,
+}: SquadMemberActionsProps): ReactElement {
+  const { showPrompt } = usePrompt();
+  const { user: loggedUser } = useAuthContext();
+  const { role, user } = member;
+
+  const onConfirmUnblock = async (e: React.MouseEvent) => {
+    e.preventDefault();
+
+    const hasUnblocked = await showPrompt({
```

Comment:
Minor thing, but think now we mainly do something like this:
```
if (await showPrompt(promptOptions)) {
```

Reply:
Ah, just my personal preference, I kinda don't like seeing `await` inside `if` conditions.

Though I think we do this when we are reusing the options, but to keep consistency let me update this one.

Reply:
Yeah either way has it's charm, maybe something we can create a rule for?